benchmark.go

```diff
@@ -112,13 +143,28 @@ func runBenchmark(out output, logger *zap.Logger, allOpts Options, resolved reso
 
 	goMaxProcs := opts.setGoMaxProcs()
 	numConns := opts.getNumConnections(goMaxProcs)
-	out.Printf("Benchmark parameters:\n")
-	out.Printf("  CPUs:            %v\n", goMaxProcs)
-	out.Printf("  Connections:     %v\n", numConns)
-	out.Printf("  Concurrency:     %v\n", opts.Concurrency)
-	out.Printf("  Max requests:    %v\n", opts.MaxRequests)
-	out.Printf("  Max duration:    %v\n", opts.MaxDuration)
-	out.Printf("  Max RPS:         %v\n", opts.RPS)
+
+	parameters := Parameters{
+		CPUs:        goMaxProcs,
+		Connections: numConns,
+		Concurrency: opts.Concurrency,
+		MaxRequests: opts.MaxRequests,
+		MaxDuration: opts.MaxDuration.String(),
+		MaxRPS:      opts.RPS,
+	}
+
+	// If format is JSON, benchmark parameters are printed after benchmark is run to maintain a single JSON blob
+	formatAsJSON := false
+	opts.Format = strings.ToLower(opts.Format)
+	if opts.Format == "json" {
+		formatAsJSON = true
+	} else if opts.Format != "" {
+		out.Printf("Unrecognized format option %q, please specify 'json' for JSON output. Printing plaintext output as default.\n\n", opts.Format)
+	}
```

Comment:
optional: can use a `switch` case to avoid mutating the `opts.Format` with a variable that's only in-scope for the `switch`, and is a little easier to read:

```go
switch format := strings.ToLower(opts.Format); format {
case "":
  // plain text
case "json":
  formatAsJSON = true
default:
  // print error
}
```